### PR TITLE
Danger is my middle name!

### PR DESCRIPTION
**Pull request by szepeviktor**
💥 Allow duplicated keys.

For #182

### CODE REVIEW

File: dev/DataTarget/EnumFile.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function writeCases(SortingInterface $sorting): self
 
         $firstMethodPos = mb_strpos($enumContent, ' public ');
         $newEnumContent = mb_substr($enumContent, 0, $startEnum + 1);
-        $cases = array_unique($this->cases);
```
Comment (PrinsFrank) [inline dev/DataTarget/EnumFile.php:96]: This indeed looks like something that is currently broken, as array_unique calls the __toString method, which only relies on the name, not the value of an enum case;
```php
$array = [
    new EnumCase('foo', 'bar'),
    new EnumCase('foo', 'bop')
];

$array2 = array_unique($array); // [new EnumCase('foo', 'bar')]
```

**Comment by PrinsFrank** [comment]
Hi, I think you found quite the bug here. As far as I can see, the __toString method on the EnumCase class is only there to filter out duplicates, but that is not done directly. IMHO we just add the duplicate cases instead of silently dropping them and let CI pick those up for manual review. I think the best fix in that case would be this:

## dev/DataTarget/EnumCase.php
```diff
public function __toString(): string
{
-    return NameNormalizer::normalize($this->name);
+    return NameNormalizer::normalize($this->name) . NameNormalizer::normalize($this->value);
}
```

## EnumCaseTest.php
```diff
<?php
declare(strict_types=1);

use PHPUnit\Framework\TestCase;
use PrinsFrank\Standards\Dev\DataTarget\EnumCase;

/** @coversDefaultClass \PrinsFrank\Standards\Dev\DataTarget\EnumCase */
class EnumCaseTest extends TestCase
{
    /** @covers ::__toString */
    public function testToStringCanBeUsedToDetectIdenticalEnumNameValuePairs(): void
    {
        static::assertEquals(
            [
                new EnumCase('', '')
            ],
            array_unique(
                [
                    new EnumCase('', '')
                ]
            )
        );
        static::assertEquals(
            [
                new EnumCase('foo', 'bar')
            ],
            array_unique(
                [
                    new EnumCase('foo', 'bar')
                ]
            )
        );
        static::assertEquals(
            [
                new EnumCase('foo', 'bar')
            ],
            array_unique(
                [
                    new EnumCase('foo', 'bar'),
                    new EnumCase('foo', 'bar')
                ]
            )
        );
        static::assertEquals(
            [
                new EnumCase('foo', 'bar'),
                new EnumCase('foo', 'bop'),
            ],
            array_unique(
                [
                    new EnumCase('foo', 'bar'),
                    new EnumCase('foo', 'bop')
                ]
            )
        );
    }
}
```

What do you think?

**Comment by PrinsFrank** [comment]
FYI: there's currently no location the EnumTestCase would make sense in, so you can move the files from `tests/Unit/` to `tests/src/Unit` and add a new `tests/dev` folder I guess?

**Review by PrinsFrank — APPROVED** [review]

**Comment by PrinsFrank** [comment]
This looks like a change we'll need anyway, so I'll merge this. Thanks!!

**Comment by szepeviktor** [comment]
You know what you do!

**Pull request merged**